### PR TITLE
Implemented citizen's dashboard and Complaint detail

### PR DESCRIPTION
**Pull request by guty**
### Description
Citizen's dashboard is a temporary view and _complaints_ details doesn't exists.

### Solution
Implemented a couple of views for _Citizen's dashboard_  and _Complaint detail_; in the dashboard it's possible to view all the complaints registered by the _current citizen (logedin)_. Besides, _notices, alerts, dates, complaint's status and Devise messages_ were translated.

### Screenshots

_Citizen's Dashboard_:
imagen

_Complaint's detail_:
imagen

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,4 +2,16 @@
 
 class ApplicationController < ActionController::Base
   include Pundit
+
+  def after_sign_in_path_for(_resource)
```
Comment (geraldamezcua) [inline app/controllers/application_controller.rb:6]: que tal si lo refactorizamos así:

```rb
  def after_sign_in_path_for(_resource)
    return :dashboards if current_user.citizen?

    :root
  end
```

File: app/controllers/complaints_controller.rb [hunk 2]
```diff
@@ -4,7 +4,14 @@ class ComplaintsController < AuthorizationsController
   before_action :set_complaint, only: %i[show edit update destroy]
   before_action :fetch_categories, only: %i[new create edit update]
 
-  def index; end
+  def index
+    @complaint = current_user.complaints.find_by(id: params[:id])
```
Comment (geraldamezcua) [inline app/controllers/complaints_controller.rb:8]: esta debería de ser la acción show 😬

File: app/helpers/complaints_helper.rb [hunk 3]
```diff
@@ -1,7 +1,21 @@
 # frozen_string_literal: true
 
 module ComplaintsHelper
+  STATUS_CSS = { created: "is-dark",
+                 in_process: "is-warning",
+                 attended: "is-info",
+                 attended_by_program: "is-info",
+                 rejected: "is-danger" }.freeze
```
Comment (geraldamezcua) [inline app/helpers/complaints_helper.rb:8]: para esto estaría chido que implementaras un `enum`

File: app/controllers/dashboards_controller.rb [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # frozen_string_literal: true
 
 class DashboardsController < AuthorizationsController
-  def index; end
+  def index
+    @complaints = current_user.complaints.all
+    @created = Complaint.where("status" => "created").count
```
Comment (geraldamezcua) [inline app/controllers/dashboards_controller.rb:6]: los enums nos generan muchos metodos :D 
y nos generan scopes para generar queries:

```rb
Complaint.created
Complaint.in_process
Complaint.attended_by_program
Complaint.rejected
```

File: app/helpers/complaints_helper.rb [hunk 5]
```diff
@@ -1,7 +1,21 @@
 # frozen_string_literal: true
 
 module ComplaintsHelper
+  STATUS_CSS = { created: "is-dark",
```
Comment (itsmeurbi) [inline app/helpers/complaints_helper.rb:4]: You don't have to worry about this right now, but constant name is singular while it represents a collection
I'd use a pluralize the name instead

File: app/helpers/dashboard_helper.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module DashboardHelper
+  def complaints_created
```
Comment (itsmeurbi) [inline app/helpers/dashboard_helper.rb:4]: Calling this method I'd expect a collection, not a number
I think a more descriptive name would be `number_of_created_complaints`
But because I see it's scoped by a user, maybe `number_of_created_complaints_by_user` IDK hehe

File: app/helpers/dashboard_helper.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+  def complaints_in_process
+    current_user.complaints.in_process.count
+  end
+
+  def complaints_attended
+    current_user.complaints.attended.count + current_user.complaints.attended_by_program.count
+  end
+
+  def complaints_rejected
```
Comment (itsmeurbi) [inline app/helpers/dashboard_helper.rb:16]: `complaints_created`, `complaints_in_process` and `complaints_rejected ` are veeeery similar
We could do something like:

```
def number_of_complaints_by_user(type:)
  current_user.complaints.where(status: type.size
end
```

Also you may find [this](https://mensfeld.pl/2014/09/activerecord-count-vs-length-vs-size-and-what-will-happen-if-you-use-it-the-way-you-shouldnt/) interesting 
TLDR: `count` always performs a SQL query, `size` will count the collection if it was previously loaded, if not it'll perform a SQL query then

File: app/helpers/formats_helper.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module FormatsHelper
+  def date_format(date)
```
Comment (itsmeurbi) [inline app/helpers/formats_helper.rb:4]: `date_format`....which format? There might be dozens of date formats 🤔 
Having a more explicit name would be better, like `short_date` or making your method to required the desired format, but this doesn't make sense, method would be bigger that what it does haha

**Review by itsmeurbi — CHANGES_REQUESTED** [review]

**Review by geraldamezcua — APPROVED** [review]
